### ISSUE TITLE
/meta/main.yml schema does not contain role argument_spec details

### ISSUE THREAD

**Issue by benwa** [ansible/ansible-lint#4762]
### Summary

When the Ansible Argument Specification file is named `/meta/main.yml`, it is not recognized as such. `/meta/argument_specs.yml` works as expected.

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#role-argument-validation

### Extension version

25.8.1

### VS Code version

1.103.2

### Ansible Version

```shell
$ ansible --version
ansible [core 2.18.6]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/bennett/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.13/site-packages/ansible
  ansible collection location = /home/bennett/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/sbin/ansible
  python version = 3.13.7 (main, Aug 14 2025, 00:00:00) [GCC 15.2.1 20250808 (Red Hat 15.2.1-1)] (/usr/bin/python3)
  jinja version = 3.1.6
  libyaml = True
```

### OS / Environment

Windows 11, WSL (Fedora 42), ansible-lint 25.8.2

### Relevant log output

```shell

```

**Comment by alisonlhart** [comment]
Hi @benwa , can you provide some more information about the issue you're seeing? Currently both argument_spec and meta/main.yml files are recognized in the extension as "Ansible" language files and I am unable to reproduce this issue as I understand it. 

If you are running a role and not getting argument spec validation with the argument spec inside the role's meta/main.yml file, that is a more fundamental Ansible issue. If it is related to Ansible language server file recognition and schema validation inside VS Code when using the extension, that is functionality on the extension side.

Please provide a reproducer so we can look further into this.

**Comment by benwa** [comment]
Basically, when the file is named `/meta/main.yml`, VSCode doesn't load the JSON specification like it does when it is named `/meta/argument_specs.yml`

<img width="1435" height="136" alt="Image" src="https://github.com/user-attachments/assets/3cf5c399-58be-40e1-b6bf-747795aa20dd" />

**Comment by alisonlhart** [comment]
@benwa Got it, thanks for confirming. This is expected behavior, since the `meta/main.yml` file _can_ include role argument specs, but it may also include other metadata about the role. In this way, it can't be validated against the dedicated argument spec schema in the same way as if it were only an argument spec file. However, the `meta/main.yml` schema we're using should include the argument_specs structure and it currently doesn't contain that info. 
I'm going to transfer this issue to ansible-lint, where the schema we use for `meta/main` schema originates.

**Comment by benwa** [comment]
Makes sense, thank you @alisonlhart!
